asm/inline.rs

```diff
@@ -20,45 +22,58 @@ pub unsafe fn __control_r() -> u32 {
 
 #[inline(always)]
 pub unsafe fn __control_w(w: u32) {
-    asm!("msr CONTROL, {}", in(reg) w);
+    // ISB is required after writing to CONTROL,
+    // per ARM architectural requirements (see Application Note 321).
+    asm!("msr CONTROL, {}", "isb", in(reg) w);
+
+    // Ensure instructions are not reordered around the CONTROL update.
+    compiler_fence(Ordering::SeqCst);
```

Comment:
This only affects *memory* instructions. Probably a good idea to clarify that in the comments.

Reply:
I've done so, but will the `asm!` block's default memory clobber ensure no memory accesses can be reordered around the block anyway? Will the compiler fence even survive being turned into an FFI call?

I've added these fences on the understanding that the compiler might otherwise reorder instructions around the `asm` call, but I don't fully see what the fence would prevent that's not prevented by the `asm!` block and/or FFI call (with or without LTO inlining).